packages/ui-time-select/src/TimeSelect/props.ts

```diff
@@ -157,61 +122,63 @@ const propTypes: PropValidators<PropKeys> = {
    *   type: PropTypes.oneOf(['error', 'hint', 'success', 'screenreader-only'])
    * }`
    */
-  messages: PropTypes.arrayOf(FormPropTypes.message),
+  messages?: FormMessage[]
   /**
    * The placement of the options list.
    */
-  placement: PositionPropTypes.placement,
+  placement?: PlacementPropValues
   /**
    * The parent in which to constrain the placement.
    */
-  constrain: PositionPropTypes.constrain,
+  constrain?: PositionConstraint
   /**
    * Callback fired when a new option is selected.
    * @param {Object} event - the event object
    * @param {Object} data - additional data
    * @param data.value - the value of selected option
+   * @param data.inputText - the value in the input text
    */
-  onChange: PropTypes.func,
+  onChange?: (
+    event: React.SyntheticEvent,
+    data: { value?: string; inputText: string }
+  ) => void
   /**
    * Callback fired when text input receives focus.
    */
-  onFocus: PropTypes.func,
+  onFocus?: (event: React.FocusEvent<HTMLInputElement>) => void
   /**
    * Callback fired when text input loses focus.
    */
-  onBlur: PropTypes.func,
+  onBlur?: (event: React.FocusEvent<HTMLInputElement>) => void
   /**
    * Callback fired when the options list is shown.
    */
-  onShowOptions: PropTypes.func,
+  onShowOptions?: (event: BaseSyntheticEvent) => void
```

Comment:
I don't think this is correct anymore: it is used in the handleInputChange method, and that is coming from Select, which was typed as `onInputChange?: ((event: React.ChangeEvent<HTMLInputElement>, value: string) => void` since then.